ui/src/features/get-last-reports.js

```diff
@@ -43,16 +43,18 @@ class getReports extends React.Component {
     };
   }
 
-    filterByFavoriteState = () => {
+    filterFavorites = () => {
       const { onlyFavorites, sortedReports } = this.state;
-      const filteredReports = sortedReports.filter((report) => (!!report.is_favorite) === onlyFavorites);
-      this.setState({ sortedReports: filteredReports, sortHeader: '' });
+      if (onlyFavorites) {
+        const filteredReports = sortedReports.filter((report) => (!!report.is_favorite) === onlyFavorites);
```

Comment:
can be changed to:
const filteredReports = sortedReports.filter((report) => (report.is_favorite));